projects/RabbitMQ.Client/client/api/IBasicConsumer.cs

```diff
@@ -86,7 +83,7 @@ public interface IBasicConsumer
         /// </summary>
         /// <remarks>
         /// Does nothing with the passed in information.
-        /// Note that in particular, some delivered messages may require acknowledgement via <see cref="IModel.BasicAck"/>.
+        /// Note that in particular, some delivered messages may require acknowledgement via <see cref="IChannel.AckMessageAsync"/>.
         /// The implementation of this method in this class does NOT acknowledge such messages.
         /// </remarks>
         void HandleBasicDeliver(string consumerTag,
```

Comment:
Will channel become mostly async? If yes we should probably be async all the way

Reply:
Totally agree. Make the HandleBasic* methods async and return ValueTask. That way they can be potentially allocation-free even in some async-cases.